core/arch/arm/mm/core_mmu.c

```diff
@@ -959,6 +959,10 @@ static void check_pa_matches_va(void *va, paddr_t pa)
 		return;
 	}
 #ifdef CFG_WITH_PAGER
+	if (v >= CFG_TEE_LOAD_ADDR && v < core_mmu_linear_map_end) {
+		TEE_ASSERT(v == pa);
```

Comment:
current assert already provides the function/line info where is asserted.
it not as explicit as a dedicated message of course.

On 5 July 2016 at 19:38, Jérôme Forissier <notifications@github.com> wrote:

> In core/arch/arm/mm/core_mmu.c
> <https://github.com/OP-TEE/optee_os/pull/899#discussion_r69605518>:
>
> > @@ -959,6 +959,10 @@ static void check_pa_matches_va(void *va, paddr_t pa)
> >  		return;
> >  	}
> >  #ifdef CFG_WITH_PAGER
> > +	if (v >= CFG_TEE_LOAD_ADDR && v < core_mmu_linear_map_end) {
> > +		TEE_ASSERT(v == pa);
>
> @jenswi-linaro <https://github.com/jenswi-linaro> +1 for fixing our
> assert() wrt. NDEBUG to properly follow the standard.
>
> I also agree that TEE_ASSERT is confusing (is the expression always
> evaluated?), and I think your suggestion to simply remove it is good. While
> we're at it, I would suggest adding a reason string, like the Linux panic
> function: if (!expr) panic("Error message"). The implementation could
> print the message only, or the message plus the file/line info, depending
> on the debug level.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/OP-TEE/optee_os/pull/899/files/87f74fa091e7f061a8d8a965c2555295d2dadb22#r69605518>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ATEHJi1ma2UmJKiPGff4lGlAz897RBsaks5qSpaYgaJpZM4JFEqR>
> .
>